### PR TITLE
RHEL Atomic

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by mvollmer** [comment]
@petervo, I need help with this...

`cockpit-ostree` requires `rpm-ostree >= 2015.10-1` but there is only `rpm-ostree-client-2015.9.2` on RHEL Atomic, as far as I can see.

Removing the requirement from `cockpit-ostree` then stops with this error in `rhel-atomic.install`:
```
error: Unknown option --timestamp=2016-01-11 11:47:26 +0000
```

**Comment by mvollmer** [comment]
@petervo says that rpm-ostree on RHEL Atomic is too old.  Should we skip the ostree feature?

**Comment by stefwalter** [comment]
> @petervo says that rpm-ostree on RHEL Atomic is too old. Should we skip the ostree feature?

@jscotka What do you think? For now, that's fine if it lets us make progress. We need to enable testing of this feature shortly so it can be tested for the next RHEL Atomic Host release.

**Comment by mvollmer** [comment]
@petervo, this error happens in rhel-atomic.install:

```
error: Unknown option --timestamp=2016-01-11 11:47:26 +0000
```

I have simply removed the `--timestamp` option from the invocation, and things still seem to work.  Can you say you added that option?

**Comment by petervo** [comment]
It was an attempt to work around a bug in ostree that was causing crashes, but it never completely eliminated them. It's ok to remove.

**Comment by mvollmer** [comment]
> It was an attempt to work around a bug in ostree that was causing crashes, but it never completely eliminated them. It's ok to remove.

Ok, thanks.

File: test/check-multi-machine [hunk 1]
```diff
@@ -87,6 +87,11 @@ class TestMultiMachine(MachineCase):
         self.machine3.wait_boot()
         self.machine3.execute("hostnamectl set-hostname machine3")
 
+        # HACK - https://bugzilla.redhat.com/show_bug.cgi?id=1297745
+        if self.machine.image == "rhel-atomic":
```
Comment (stefwalter) [inline test/check-multi-machine]: Would prefer a naughty file so we can track when this issue goas away. Is it possible to implement that way?

File: test/testlib.py [hunk 2]
```diff
@@ -586,6 +590,10 @@ def login_and_go(self, path=None, user=None, host=None):
         "(audit: )?type=1403 audit.*",
         "(audit: )?type=1404 audit.*",
 
+        # Happens on rhel-atomic
+        "type=1400 .*granted.*comm=\"tuned\".*",
```
Comment (stefwalter) [inline test/testlib.py]: Need a bug file for this one.

File: test/testlib.py [hunk 3]
```diff
@@ -586,6 +590,10 @@ def login_and_go(self, path=None, user=None, host=None):
...
+        "type=1400 .*denied.*comm=\"iptables\".*name=\"xtables.lock\".*",
```
Comment (stefwalter) [inline test/testlib.py]: This is https://bugzilla.redhat.com/show_bug.cgi?id=1296826

File: test/vm-create [hunk 4]
```diff
@@ -73,6 +73,10 @@ class MachineBuilder:
             self.machine.upload([ "guest/lib" ], "/var/lib/testvm")
             self.machine.upload([script], "/var/tmp/SETUP")
             self.machine.upload([source], "/var/tmp")
+
+            if "rhel" in self.machine.image:
+                self.machine.upload([ "/home/mvo/.rhel" ], "/root/")
```
Comment (stefwalter) [inline test/vm-create]: Code accidentally left in?

File: test/check-multi-machine (hunk 1 above)
Comment (mvollmer) [inline test/check-multi-machine]: There is a lot of fallout from the non-writable /tmp and we would have many known issues.  (Maybe 5 or 6?)  I would argue that this isn't a work around, but the real fix (just in the wrong place), and with it in place we can already now run more of our tests.

File: test/vm-create (hunk 4 above)
Comment (mvollmer) [inline test/vm-create]: No, but forgot to use $HOME instead of hard-coding ~mvo.  Ouch.

File: test/check-multi-machine (hunk 1 above)
Comment (stefwalter) [inline test/check-multi-machine]: Yes, there are tradeoffs. But in reality this is broken for real users. We don't want Cockpit to show as green for a given OS, and then have those features fail in real life.

Although on the other hands, users do have an easy work around ... after each boot. So it's not cut and dry here.

File: test/vm-create (hunk 4 above)
Comment (mvollmer) [inline test/vm-create]: Fixed.

File: test/check-multi-machine (hunk 1 above)
Comment (mvollmer) [inline test/check-multi-machine]: > Yes, there are tradeoffs. But in reality this is broken for real users. We don't want Cockpit to show as green for a given OS, and then have those features fail in real life.

Very true, and I was saying that I didn't want to do workarounds anymore, didn't I?  I'll create the nauhgty patterns.

File: test/testlib.py (hunk 2 above)
Comment (mvollmer) [inline test/testlib.py]: Done: https://bugzilla.redhat.com/show_bug.cgi?id=1298157

File: test/testlib.py (hunk 3 above)
Comment (mvollmer) [inline test/testlib.py]: Interestingly, this is a race during boot.  If kube is started early enough after boot, the messages appear.  If we wait 5 seconds before starting kube, they don't appear.

Comment (mvollmer) [inline test/testlib.py]: https://bugzilla.redhat.com/show_bug.cgi?id=1298171

File: test/check-multi-machine (hunk 1 above)
Comment (mvollmer) [inline test/check-multi-machine]: Done.

**Pull request closed without merging**